src/Algebra/Properties/IdempotentCommutativeMonoid.agda

```diff
@@ -0,0 +1,38 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some derivable properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (IdempotentCommutativeMonoid)
+
+module Algebra.Properties.IdempotentCommutativeMonoid
+  {c ℓ} (M : IdempotentCommutativeMonoid c ℓ) where
+
+open IdempotentCommutativeMonoid M
+
+open import Algebra.Consequences.Setoid setoid
+  using (comm∧distrˡ⇒distrʳ; comm∧distrˡ⇒distr)
+open import Algebra.Definitions _≈_
+  using (_DistributesOverˡ_; _DistributesOverʳ_; _DistributesOver_)
+open import Algebra.Properties.CommutativeSemigroup commutativeSemigroup
+  using (interchange)
+open import Relation.Binary.Reasoning.Setoid setoid
+
+
+------------------------------------------------------------------------
+-- Distributivity
+
+distrˡ : _∙_ DistributesOverˡ _∙_
```

Comment:
I'm wondering whether we should make it explicit in the name that the operation is distributing over itself, i.e. `∙-distrˡ-∙`? It's somewhat unusual to have an operator related to itself, so it might reduce user surprise and increase readability?

Reply:
Pro: the argument you offer
Contra: `IdempotentCommutativeMonoid` only has 1 binary operation, so (albeit implicitly) there is only one 'distributivity' available/expressible?

As for relating operators to themselves, `interchange : Interchangable _∙_ _∙_` under `CommutativeSemigroup` provides a counter-example to your suggested naming, unless you think that that, too, should be `∙-interchange-∙`?

The originating issue/PR #2403 / #2407 previously had `distr` for what was clearly a *left* distributivity, so all I had done was rectify the orientation of that name, and then flesh out the right- version and their conjunction...

But it's not (especially) a deal-breaker for me either way which name/naming style we go for, but consistency seems more important in the long run. I suppose that *abstract* structures eg. `IsSemiring` don't mention their operators in their property names, but concrete instances for eg `Nat` do... and this is an 'abstract' property... should it in fact not be here at all, but be a manifest field of the `IsIdempotentCommutativeMonoid` structure instead?

Reply:
Another (possible) contra: #1544 ?

Reply:
Anyway: pushed now (but leaving `interchange` alone, for the time being... ;-))

Reply:
> As for relating operators to themselves, interchange : Interchangable _∙_ _∙_ under CommutativeSemigroup provides a counter-example to your suggested naming, unless you think that that, too, should be ∙-interchange-∙?

Yes, I admit it's inconsistent at the moment. I'm not aware we have a naming convention for interchange yet, I would probably argue for something like `∙-∙-interchange`

>  I suppose that abstract structures eg. IsSemiring don't mention their operators in their property names, but concrete instances for eg Nat do... and this is an 'abstract' property... 

They do when there's potential ambiguity. 

> should it in fact not be here at all, but be a manifest field of the IsIdempotentCommutativeMonoid structure instead?

I don't think we can add every derived property as a manifest field otherwise the records are going to bloat like crazy.